code/game/objects/items/granters/crafting/pipegun.dm

```diff
@@ -1,10 +1,16 @@
-/obj/item/book/granter/crafting_recipe/pipegun_prime
+/obj/item/book/granter/crafting_recipe/maint_gun
+	icon_state = "book1"
+
+/obj/item/book/granter/crafting_recipe/maint_gun/recoil(mob/living/user)
```

Comment:
As above with the component, this is another case of "good idea to make this a subtype but you gave it a name which is too specific".
If all this is doing is sharing the "turn to ash" behaviour there's no reason to name it specifically for maintenance guns, we might want to use it for things which are not guns nor found in maintenance.